torch/_inductor/ir.py

```diff
@@ -3845,20 +3846,19 @@ def __init__(self, *, kernel_idx, grid, kernel_args):
         self.name = V.graph.register_buffer(self)
         self.kernel_idx = kernel_idx
         self.grid = grid
-
-    @classmethod
-    def create(cls, *, kernel_idx, grid, kernel_args):
-        packed = UserDefinedTritonKernel(
-            kernel_idx=kernel_idx, grid=grid, kernel_args=kernel_args
-        )
-        for arg in kernel_args.values():
-            if isinstance(arg, TensorBox):
-                MutationOutput(arg.layout, arg, packed)
+        mark_node_as_mutating(self, *kernel_args.values())
 
     def get_alias_names(self):
         return [i.get_name() for i in self.inputs]
 
 
+def mark_node_as_mutating(cur_buffer, *mutated_ops):
+    for op in mutated_ops:
+        if isinstance(op, TensorBox):
```

Comment:
Why will this be called with non-TensorBox inputs?  Could be a risk of silent correctness errors since this will silently ingore incorrect inputs.